starport/services/scaffolder/component.go

```diff
@@ -142,3 +147,72 @@ func isGoReservedWord(name string) bool {
 	}
 	return false
 }
+
+// isComponentCreated checks if the component has been already created with Starport in the project
+func isComponentCreated(appPath, moduleName, compName string) (bool, error) {
+	compNameTitle := strings.Title(compName)
+	typesToCheck := []string{
+		compNameTitle,
+		"MsgCreate" + compNameTitle,
+		"MsgUpdate" + compNameTitle,
+		"MsgDelete" + compNameTitle,
+		"Msg" + compNameTitle,
+		"Query" + compNameTitle + "Request",
+		"Query" + compNameTitle + "Response",
+		"QueryAll" + compNameTitle + "Request",
+		"QueryAll" + compNameTitle + "Response",
+		"QueryGet" + compNameTitle + "Request",
+		"QueryGet" + compNameTitle + "Response",
+		"MsgSend" + compNameTitle,
+		compNameTitle + "PacketData",
+	}
+
+	return checkTypesDefined(appPath, moduleName, typesToCheck)
+}
+
+// checkTypesDefined returns true if at least one of the provided type is defined in the type package of the module
+func checkTypesDefined(appPath, moduleName string, typeNames []string) (exist bool, err error) {
```

Comment:
i think it would be nice if this function could provide more context on the conflict, e.g. "conflicts with MsgFoo".
i would provide this context with a custom error, like fmt.Errorf("%w: conflicts %s", ErrValidationConflict, objType), and maybe even remove a boolean from a return signature

Reply:
Yes it is a good idea